packages/core/src/model.js

```diff
@@ -1694,6 +1685,30 @@ ${associationOwner._getAssociationDebugList()}`);
     options.offset = null;
     options.order = null;
 
+    // counting grouped rows is not possible with `this.aggregate`
+    // use a subquery to get the count
+    if (options.group && options.countGroupedRows) {
+      // remove trailing semicolon for wrap query
+      const query = this.queryGenerator.selectQuery(this.getTableName(), options).slice(0, -1);
```

Comment:
```suggestion
      const query = removeTrailingSemicolon(this.queryGenerator.selectQuery(this.table, options));
```

You'll need to import `removeTrailingSemicolon`;
https://github.com/sequelize/sequelize/blob/89fc0ff0dec093d17cd37d2f0260e404926faeb0/packages/core/src/utils/string.ts#L109-L115

`getTableName()` is deprecated as per;
https://github.com/sequelize/sequelize/blob/89fc0ff0dec093d17cd37d2f0260e404926faeb0/packages/core/src/model-typescript.ts#L413-L433

Reply:
I tried to keep everything from the previous PR. So 
I was also curious of some of the issues you mentioned. 
I will try to go re-review all the codes!

Reply:
resolved at https://github.com/sequelize/sequelize/pull/17751/commits/b761e8f643f1ddf2896469d79680cec8d545e6d6